### PR TITLE
Feat: Add display_in_post option for photos

### PR DESCRIPTION
**Pull request by xu-song**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [ ] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [x] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

In the current theme, `preview_photo` is **locked at the top** of post.

This commit frees the location of `preview_photo`.

## What is the new behavior?

* Screens with this changes: 

**preview page**

**post page**

<table border="1"><colgroup><col width="30%"><col width="70%"></colgroup><tbody><tr><td></td><td></td></tr></tbody></table>

Obviously, the right one is more flexible.

* Link to demo site with this changes:  http://blog.eson.org/

### How to use?

In Post
```
---
title: abc
photos:
  - file_path.jpg
---

...
```

In NexT `_config.yml`:
```yml
# Post photo preview settings. Set photos in post front-matter first
photos:
  display_in_post: false
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: layout/_macro/post.swig [hunk 1]
```diff
@@ -283,7 +283,7 @@
     <div class="post-body{% if theme.han %} han-init-context{% endif %}{% if post.direction && post.direction.toLowerCase() === 'rtl' %} rtl{% endif %}" itemprop="articleBody">
 
       {# Gallery support #}
-      {% if post.photos and post.photos.length %}
+      {% if post.photos and post.photos.length and (theme.photos.display_in_post or is_index) %}
```
Comment (sli1989) [inline layout/_macro/post.swig]: but u need to add cover photo in post manually?

Comment (xu-song) [reply layout/_macro/post.swig]: yes

**Comment by stevenjoezhang** [comment]
Why not design it as an option in post front matter instead of `_config.yml`?

**Comment by xu-song** [comment]
It is first design for general config which works for all posts.

It is also a good idea to set in post front matter. such as
```
---
title: abc
photos:
  - file_path.jpg
display_in_post: true
---
```

or any other suggestion

**Comment by ivan-nginx** [comment]
@xu-song recreate this pull, if u still want's to merge. Thank's.

**Comment by stevenjoezhang** [comment]
Need to reconsider post gallery feature

**Pull request closed without merging**